samza-core/src/main/scala/org/apache/samza/system/chooser/BootstrappingChooser.scala

```diff
@@ -103,8 +112,8 @@ class BootstrappingChooser(
 
   def start = {
     // remove the systemStreamPartitions not registered.
-    laggingSystemStreamPartitions = laggingSystemStreamPartitions.filter(registeredSystemStreamPartitions.contains(_))
-    systemStreamLagCounts = laggingSystemStreamPartitions.groupBy(_.getSystemStream).map {case (systemStream, ssps) => systemStream -> ssps.size}
+    laggingSystemStreamPartitions = laggingSystemStreamPartitions.filter((sspToTaskCount: (SystemStreamPartition, Int)) => registeredSystemStreamPartitions.contains(sspToTaskCount._1))
+    systemStreamLagCounts = laggingSystemStreamPartitions.keys.groupBy(_.getSystemStream).map {case (systemStream, ssps) => systemStream -> ssps.size}
```

Comment:
please add comment why do we have 2 counters and what is the relationship between them.